syntax = "proto3";

package dynamic_configurations;

option go_package = "dynamic_configurations.v1;v1";

service DynamicConfigurations {
    rpc Put(PutRequest) returns (PutResponse) {}
    rpc Get(GetRequest) returns (GetResponse) {}
}

message PutRequest {
    string key = 1;
    bytes value = 2;
}

message PutResponse {
    int64 requestID = 1;
    string timestamp = 2;
    float executionTime = 3;
    string message = 4;
}

message GetRequest {
    string key = 1;
}

message GetResponse {
    int64 requestID = 1;
    string timestamp = 2;
    float executionTime = 3;
    string message = 4;
    bytes value = 5;
}